doc/modules/model_evaluation.rst

```diff
@@ -2354,6 +2354,24 @@ the difference in errors decreases. Finally, by setting, ``power=2``::
 we would get identical errors. The deviance when ``power=2`` is thus only
 sensitive to relative errors.
 
+.. _d2_tweedie_score:
+
+D² score, the coefficient of determination
+-------------------------------------------
+
+The :func:`d2_tweedie_score` function computes the percentage of deviance
+explained. It is a generalization of R², where the squared error is replaced by
+the Tweedie deviance. D², also known as McFadden's likelihood ratio index, is
+calculated as
+
+.. math::
+
+  D^2(y, \hat{y}) = 1 - \frac{\text{D}(y, \hat{y})}{\text{D}(y, \bar{y})} \,.
+
+The argument `power` defines the Tweedie power as for
+:func:`mean_tweedie_deviance`. Note that for `power=0`,
+:func:`d2_tweedie_score` equals :func:`r2_score` (for single targets).
+
```

Comment:
Maybe we should add some quick notes on how to interpret this score, for instance:

Higher values means the model predictions are a better estimate of the expected mean of a conditional Tweedie random variable. D² is upper bounded by 1.0 (perfect predictions).

Values close to zero or negatives are typically achieved by a model that predicts no better than an constant predictor that predict would the empirical mean of the test set and ignores the input features.

Reply:
I just noticed you wrote something similar but more concisely in the docstring. I think it's worth repeating it here in the narrative doc.

Reply:
Also shall we add a code snippet to show how to use it with `make_scorer` to define a scorer object to be used in grid search cv and co for a given value of `power`?